homeassistant/components/camera/__init__.py

```diff
@@ -0,0 +1,273 @@
+
+    hass.http.register_path(
+        'GET',
+        re.compile(r'/api/camera_proxy/(?P<entity_id>[a-zA-Z\._0-9]+)'),
+        _proxy_camera_image,
+        require_auth=True)
+
+    # pylint: disable=unused-argument
+    def _proxy_camera_mjpeg_stream(handler, path_match, data):
+        """ Proxies the camera image as an mjpeg stream via the HA server.
+        This function takes still images from the IP camera and turns them
+        into an MJPEG stream.  This means that HA can return a live video
+        stream even with only a still image URL available.
+        """
+        entity_id = path_match.group(ATTR_ENTITY_ID)
+
+        camera = None
+        if entity_id in component.entities.keys():
+            camera = component.entities[entity_id]
+
+        if camera:
+
+            try:
+                camera.is_streaming = True
+                camera.update_ha_state()
+
+                handler.request.sendall(bytes('HTTP/1.1 200 OK\r\n', 'utf-8'))
+                handler.request.sendall(bytes(
+                    'Content-type: multipart/x-mixed-replace; \
+                        boundary=--jpgboundary\r\n\r\n', 'utf-8'))
+
+                handler.request.sendall(bytes('--jpgboundary\r\n', 'utf-8'))
+
+                while True:
+
+                    if camera.username and camera.password:
+                        response = requests.get(
+                            camera.still_image_url,
+                            auth=HTTPBasicAuth(
+                                camera.username,
+                                camera.password))
+                    else:
+                        response = requests.get(camera.still_image_url)
+
+                    headers_str = '\r\n'.join((
+                        'Content-length: {}'.format(len(response.content)),
+                        'Content-type: image/jpeg',
+                    )) + '\r\n\r\n'
+
+                    handler.request.sendall(
+                        bytes(headers_str, 'utf-8') +
+                        response.content +
+                        bytes('\r\n', 'utf-8'))
+
+                    handler.request.sendall(
+                        bytes('--jpgboundary\r\n', 'utf-8'))
+
+            except (requests.RequestException, IOError):
+                camera.is_streaming = False
+                camera.update_ha_state()
+
+        else:
+            handler.send_response(HTTP_NOT_FOUND)
+
+        camera.is_streaming = False
+
+    hass.http.register_path(
+        'GET',
+        re.compile(
+            r'/api/camera_proxy_stream/(?P<entity_id>[a-zA-Z\._0-9]+)'),
+        _proxy_camera_mjpeg_stream,
+        require_auth=True)
+
+
+class Camera(Entity):
+    """ The base class for camera components """
+
+    @property
+    # pylint: disable=no-self-use
+    def is_recording(self):
+        """ Returns true if the device is recording """
+        return False
+
+    @property
+    # pylint: disable=no-self-use
+    def is_streaming(self):
```

Comment:
Since this is managed by the Home Assistant core, I think we should implement these methods instead of leaving it up to the camera. Maybe this even can become a variable instead of a property ?